commit be28089dc4c1b07d9def45a3c763f432ae8322c4
Author: Vincent Untz <vuntz@gnome.org>
Date:   Mon Aug 9 10:31:45 2010 -0500

    build: fix build with glib >= 2.25.12 (bgo #626421)
    
    Work around an API break in glib.

---

diff --git a/libqcdm/tests/test-qcdm.c b/libqcdm/tests/test-qcdm.c
index 8685080..4e6f0cf 100644
--- a/libqcdm/tests/test-qcdm.c
+++ b/libqcdm/tests/test-qcdm.c
@@ -28,7 +28,11 @@ typedef struct {
     gpointer com_data;
 } TestData;
 
+#if GLIB_CHECK_VERSION(2,25,12)
+typedef GTestFixtureFunc TCFunc;
+#else
 typedef void (*TCFunc)(void);
+#endif
 
 #define TESTCASE(t, d) g_test_create_case (#t, 0, d, NULL, (TCFunc) t, NULL)
 
diff --git a/src/tests/test-modem-helpers.c b/src/tests/test-modem-helpers.c
index 92a7af8..74d2d94 100644
--- a/src/tests/test-modem-helpers.c
+++ b/src/tests/test-modem-helpers.c
@@ -791,7 +791,11 @@ test_data_free (TestData *data)
 }
 
 
+#if GLIB_CHECK_VERSION(2,25,12)
+typedef GTestFixtureFunc TCFunc;
+#else
 typedef void (*TCFunc)(void);
+#endif
 
 #define TESTCASE(t, d) g_test_create_case (#t, 0, d, NULL, (TCFunc) t, NULL)
 
